### PR TITLE
Stop Log Replication if Plugin fails to read the configuration.

### PR DESCRIPTION
**Pull request by pankti-m**
Description:
Log Replication should stop if the plugin fails to fetch important info
such as streams to replicate and version.

Why should this be merged: Better Error Handling

### CODE REVIEW

**Review by annym — APPROVED** [review]

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java [hunk 1]
```diff
@@ -319,30 +320,34 @@ private void connectToClusterManager() {
...
+            // Unblock server initialization & register to Log Replication Lock, to attempt lock / leadership acquisition
+            serverCallback.complete(interClusterReplicationService);
 
-        logReplicationEventListener = new LogReplicationEventListener(this);
-        logReplicationEventListener.start();
-        serverStarted = true;
+            logReplicationEventListener = new LogReplicationEventListener(this);
+            logReplicationEventListener.start();
+            serverStarted = true;
+        } catch (Throwable t) {
```
[the remark this answers is not in the document]
Comment (zhangn49) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java:348]: +1 We can let the upper-level function catch it.

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java [hunk 2]
```diff
@@ -319,30 +320,34 @@ private void connectToClusterManager() {
      * - Start Log Replication Server (receiver component)
      */
     private void bootstrapLogReplicationService() {
-        // Through LogReplicationConfigAdapter retrieve system-specific configurations (including streams to replicate)
-        LogReplicationConfig logReplicationConfig = getLogReplicationConfiguration(getCorfuRuntime());
+        try {
+            // Through LogReplicationConfigAdapter retrieve system-specific configurations (including streams to replicate)
```
Comment (zhangn49) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java:324]: Do we also retrieve the version info through the adapter? If so, let's modify the comment a little bit

**Review by zhangn49 — APPROVED** [review]

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (pankti-m) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java:348]: Done.

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java (hunk 2 above)
Comment (pankti-m) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java:324]: Done

**Review by zhangn49 — APPROVED** [review]

**Pull request merged**